### DISCUSSION TITLE
how can i use library osg in webAssembly?

### DISCUSSION THREAD

**Discussion by alihitman99** [openscenegraph/OpenSceneGraph#1337] [General]
i my project i use the library osg and developed with qt 6.5. and now i want to use this project for webAssembly. os is centos8. In this link, it is said that you should build OSG with emCmake or emconfigure.

[](https://github.com/cubicool/osg-wasm
)
But it gives an error when it is build. Where is the problem?

`emconfigure cmake . -DCMAKE_TOOLCHAIN_FILE=../emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake
`

**Comment by xarray** [comment]
Have a look at osgverse (https://github.com/xarray/osgverse). It supports compiling OSG and osgEarth to WebAssembly using a Setup.sh shell script, with supports of WebGL 1 and WebGL 2.

**Reply by alihitman99** [reply]
Ok thanks. i can build the OSG for web with setup.sh option 3 but i don't know how to use it for Qt QML.
https://github.com/alihitman99/osg-qml
